common/cache/metricsScopeCache.go

```diff
@@ -59,16 +61,19 @@ func NewDomainMetricsScopeCache() DomainMetricsScopeCache {
 		},
 		closeCh:       make(chan struct{}),
 		flushDuration: flushBufferedMetricsScopeDuration,
+		timeSource:    clock.NewRealTimeSource(),
 	}
 
 	mc.cache.Store(make(metricsScopeMap))
 	return mc
 }
 
 func (c *domainMetricsScopeCache) flushBufferedMetricsScope(flushDuration time.Duration) {
+	timer := c.timeSource.NewTimer(flushDuration)
+	defer timer.Stop()
 	for {
 		select {
-		case <-time.After(flushDuration):
+		case <-timer.Chan():
```

Comment:
this is probably a near-perfect candidate for a ticker instead, though it does change the semantics slightly

Reply:
ticker makes sense for metric reporting loops.